Source/WebKit/NetworkProcess/NetworkProcess.cpp

```diff
@@ -2420,7 +2420,15 @@ void NetworkProcess::websiteDataOriginDirectoryForTesting(PAL::SessionID session
     if (!session)
         return completionHandler({ });
 
-    auto clientOrigin = WebCore::ClientOrigin { WebCore::SecurityOriginData::fromURL(topOrigin), WebCore::SecurityOriginData::fromURL(origin) };
+    auto fromURL = [](const auto& url) {
+        return SecurityOriginData {
+            url.protocol().isNull() ? emptyString() : url.protocol().convertToASCIILowercase(),
+            url.host().isNull() ? emptyString() : url.host().convertToASCIILowercase(),
+            url.port()
+        };
+    };
+
+    auto clientOrigin = WebCore::ClientOrigin { fromURL(topOrigin), fromURL(origin) };
```

Comment:
I agree that we should use `WKSecurityOrigin`, but it is difficult because all of the methods available for constructing a WKSecurityOrigin (`WKSecurityOriginCreate*`) wrap around `WebCore::SecurityOrigin::create*` and those `SecurityOrigin::create` functions includes opaque origin checks (which we want to avoid).

I added a new converting constructor for `SecurityOriginData`, where that constructor directly converts `URL->SecurityOriginData`. `SecurityOriginData::fromURL` checks whether the URL should create an opaque SecurityOriginData. I believe I audited all of the `SecurityOriginData::fromURL` call sites and made appropriate changes to either directly compare URLs with `protocolHostAndPortAreEqual` or construct `SecurityOriginData(URL)`.